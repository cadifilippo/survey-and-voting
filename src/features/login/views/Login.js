import styles from './Login.module.css';
import { GithubIcon } from '../../../assets';
import { useAuth } from '../../../context/auth';
import { useEffect } from 'react';

const Login = () => {
  const { login, reload } = useAuth();

  useEffect(() => {
    window.addEventListener('hashchange', () => reload());
  }, []);

  return (
    <section className={styles.login}>
      <div className={styles.form}>
        <h1 className={styles.title}>Survey & Voting</h1>
        <button className={styles.button} onClick={login} data-testid="github">
          <GithubIcon width={40} height={40} />
          Sign In with Github
        </button>
      </div>
    </section>
  );
};

export default Login;
