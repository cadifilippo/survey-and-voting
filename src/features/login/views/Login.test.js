import { expect, test } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Login from './Login';

describe('[LOGIN] [VIEWS] Login', () => {
  test('render correctly', () => {
    const tree = render(<Login />);
    expect(tree).toMatchSnapshot();
  });

  test('should have a github button', () => {
    render(<Login />);
    expect(screen.getByTestId('github')).toBeInTheDocument();
  });
});
