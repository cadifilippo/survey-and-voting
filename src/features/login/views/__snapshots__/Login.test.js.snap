// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LOGIN] [VIEWS] Login render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="login"
      >
        <div
          class="form"
        >
          <h1
            class="title"
          >
            Survey & Voting
          </h1>
          <button
            class="button"
            data-testid="github"
          >
            <svg
              height="40"
              width="40"
            >
              github.svg
            </svg>
            Sign In with Github
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="login"
    >
      <div
        class="form"
      >
        <h1
          class="title"
        >
          Survey & Voting
        </h1>
        <button
          class="button"
          data-testid="github"
        >
          <svg
            height="40"
            width="40"
          >
            github.svg
          </svg>
          Sign In with Github
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
